packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -52,10 +51,45 @@ const toJsonBKey = <T extends keyof NestedFilterQueryFields>(nestedField: T): Js
   return [field, value].join(JSONB_FIELD_DELIMITER) as JsonBKey<T>;
 };
 
+// Inspired by Google Analytics filter: https://developers.google.com/analytics/devguides/reporting/core/v3/reference?hl=en#filters
+const operatorToSqlComparator = {
+  '==': null, // Exact match
+  '!=': '!=', // Does not match
+  '=@': 'ilike', // Contains sub string
+};
```

Comment:
Couple things about the types here:
1. `'==': '=', // Exact match` would also work right? I have a slight preference towards that because it reads a bit more clearly to me, and because typescript doesn't let you mess around with 'falsey' values as much as JS. ie. `null` != `undefined` 
2. Currently the type of this object is:
```
{
  '==': null;
  '!=': string;
  '=@': string;
};
```
The types of the keys are string literals (https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) whereas the values are just string (or null for the null one). This is how typescript works by default, but it does have it's drawbacks. 

For instance, later on you are trying to find the comparator for the operator by looking up in this table. Unfortunately the type will always be `string | null` at that point, which isn't really what you want. Instead you want the type to be `'!=' | 'ilike' | null` right?

Luckily there's a way to get this working in typescript as well!
```suggestion
const operatorToSqlComparator = {
  '==': '=' as const, // Exact match (switched to '=' here as it's my preference but if you stick with null theres no need for the 'as const')
  '!=': '!=' as const, // Does not match
  '=@': 'ilike' as const, // Contains sub string
};
```
This changes the type of the object to:
```
{
  '==': '=';
  '!=': '!=';
  '=@': 'ilike';
};
```
Which allows you to do typesafe lookups against this object!

Reply:
The reason I had `null` was to avoid having the advanced criteria if the operator is `==`. So I still prefer the filter object to be:
```
{
     type: 'district'
}
```
instead of always
```
{
     type: {
           comparator: '=',
           comparisonValue: 'district'
     }
}
```

However, I've just realized I didn't add the logic to follow the 1st format mentioned above (I was working on this quite late so that might explain 😄 ). I've followed your `=` suggestion, but still have an if check for `=` operator to avoid using comparator/comparisonValue if possible